File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static int[][] arr;
    static boolean[] node;
    static int N, V;
    static Queue<Integer> Q = new LinkedList<>();
    static StringBuilder sb = new StringBuilder();


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());
        arr = new int[N + 1][N + 1];
        node = new boolean[N + 1];

        for (int i = 0; i < M; i++) {
            StringTokenizer st_2 = new StringTokenizer(br.readLine());

            int a = Integer.parseInt(st_2.nextToken());
            int b = Integer.parseInt(st_2.nextToken());

            arr[a][b] = 1;
            arr[b][a] = 1;
        }

        dfs(V);

        node = new boolean[N+1];
        sb.append("\n");

        bfs(V);

        System.out.println(sb);

    }

    static void dfs(int V) {
        if (node[V]) return;

        node[V] = true;
        sb.append(V + " ");

        for (int i = 1; i <= N; i++) {
            if (arr[V][i] == 1) {
                dfs(i);
            }
        }
    }

    static void bfs(int V) {
        Q.offer(V);

        while (!Q.isEmpty()) {
            int len = Q.size();

            for (int i = 0; i < len; i++) {

                int v = Q.poll();

                if(!node[v]) {
                    node[v] = true;
                    sb.append(v + " ");
                }
                for (int j = 1; j <= N; j++) {
                    if (arr[v][j] == 1 && !node[j]) {
                        Q.offer(j);
                    }
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.kt

import java.util.StringTokenizer

private val bw = System.`out`.bufferedWriter()

fun main() = with(System.`in`.bufferedReader()) {
    val line = readLine().split(" ").map { it.toInt() }
    val N = line[0] // 정점의 개수
    val M = line[1] // 간선의 개수
    val V = line[2] // 시작 정점

    val graph = Array(N + 1) { ArrayList<Int>() } // 1부터 시작

    repeat(M) {
        val line = StringTokenizer(readLine())
        val start = line.nextToken().toInt()
        val end = line.nextToken().toInt()
        graph[start].add(end)
        graph[end].add(start)
    }

    for (i in 1..N) {
        graph[i].sort()
    }

    val dfsVisited = BooleanArray(N + 1)
    dfsVisited[V] = true

    fun dfs(depth: Int, node: Int) {
        if (depth == N + 1) {
            return
        }
        bw.append("$node ")

        for (nextNode in graph[node]) {
            if (!dfsVisited[nextNode]) {
                dfsVisited[nextNode] = true

                dfs(depth + 1, nextNode)
            }
        }
    }

    fun bfs() {
        val queue = ArrayDeque<Int>()
        val visited = BooleanArray(N + 1)
        queue.addLast(V)
        visited[V] = true

        while (!queue.isEmpty()) {
            val currentNode = queue.removeFirst()
            bw.append("$currentNode ")

            for (nextNode in graph[currentNode]) {
                if (visited[nextNode]) continue
                queue.addLast(nextNode)
                visited[nextNode] = true
            }
        }
    }
    dfs(1, V)

    bw.append("\n")

    bfs()

    bw.flush()
}